sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js

```diff
@@ -0,0 +1,137 @@
+/*jslint browser: true, undef: true *//*global Ext,Slate*/
+Ext.define('Slate.cbl.view.student.DashboardController', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.slate-cbl-student-dashboard',
+    requires: [
+        'Slate.cbl.model.Student',
+        'Slate.cbl.model.Competency',
+        'Slate.cbl.view.student.skill.OverviewWindow'
+    ],
+
+    config: {
+        id: 'slate-cbl-teacher-dashboard', // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+        control: {
+            '#': {
+//                contentareachange: 'refresh',
+//                progressrowclick: 'onProgressRowClick',
+                democellclick: 'onDemoCellClick',
+                render: 'onComponentRender'
+            }
+        }
+    },
+    
+    // lifecycle overrides
+    init: function() {
+        var me = this;
+
+        Ext.create('Ext.data.Store', {
+            storeId: 'cbl-competencies-loaded',
+            model: 'Slate.cbl.model.Competency'
+        });
+    },
+
+
+    //event handlers
+    onComponentRender: function(dashboardView) {
+        var me = this,
+            siteEnv = window.SiteEnvironment || {},
+            student = siteEnv.cblStudent && Ext.create('Slate.cbl.model.Student', siteEnv.cblStudent)
```

Comment:
It's not above the var because siteEnv isn't declared then, and any other configuration would reduce the compressibility. While your suggestion is clearer, the reduced explicitness is compensated for by this being an extremely common pattern.

The argument that we're doing a lot of work before the bail on line 44 would normally justify sacrificing compressibility to bail earlier before all these declarations, but in this case it would be exceptional for either student or contentArea to not be defined, in the normal user workflows they should both always be defined so this is just a graceful bailout.